test/common/router/router_ratelimit_test.cc

```diff
@@ -513,8 +513,105 @@ TEST_F(RateLimitPolicyEntryTest, DynamicMetaDataMatch) {
               testing::ContainerEq(descriptors_));
 }
 
+TEST_F(RateLimitPolicyEntryTest, MetaDataMatchDynamicSourceByDefault) {
```

Comment:
@mattklein123 I am getting an exception in the `DynamicMetaDataMatch` testcase because of the deprecation of DynamicMetaData. See: https://dev.azure.com/cncf/envoy/_build/results?buildId=54341&view=logs&j=e969334a-0e55-5c18-ac96-8b546753391e&t=b5329a9e-93c7-5412-9f4c-6f8d659e1b57&l=4042
I think keeping that test would validate that the new implementation still works with the old way of configuring it. Do you have any ideas how to overcome that exception?
The other deprecation exceptions were an oversight and fixed.

Reply:
See `DEPRECATED_FEATURE_TEST`

/wait